internal/profile/db.sql

```diff
@@ -1,5 +1,25 @@
-
-CREATE TABLE profiles (
+CREATE TABLE IF NOT EXISTS profiles (
     user_id varchar(255) PRIMARY KEY,
     profile json NOT NULL
 );
+
+ALTER TABLE profiles ADD COLUMN created_at  timestamp DEFAULT now();
+ALTER TABLE profiles ADD COLUMN updated_at  timestamp DEFAULT now();
+ALTER TABLE profiles ADD COLUMN version INTEGER DEFAULT 1;
+
+
+CREATE OR REPLACE FUNCTION update_metadata()
+RETURNS TRIGGER AS $$
+BEGIN
+  NEW.updated_at = NOW();
+  NEW.version := OLD.version + 1;
```

Comment:
I don't think this is thread safe, I was talking about a real psql sequence, so you can do next_val('seqname')